### DISCUSSION TITLE
Help with dkim signing mail

### DISCUSSION THREAD

**Discussion by GrassBlock1** [docker-mailserver/docker-mailserver#4192] [Q&A] [answered]
I have setup docker-mailserver with podman(-compose) and the default configuration (changed a bit to suit the needs) described in [Documentation](https://docker-mailserver.github.io/docker-mailserver/latest/usage/) and setup dkim with a `dkim` selector.
I have checked DNS seemed to be right, but when I test my email with mail-tester.com, the email seems always not to be signed with dkim keys.
I checked the podman logs, it says:
```
2024-09-23 10:27:43+08:00 DEBUG start-mailserver.sh: Starting opendkim
2024-09-23T11:47:42.492674+08:00 mail opendkim[178]: 6BF4F401EAE: no signing table match for 'email@domain.ltd
2024-09-23T11:47:42.493457+08:00 mail opendkim[178]: 6BF4F401EAE: no signature data
```
I have searched a bit about this problem, but no luck. Any idea for this?

**Comment by GrassBlock1** [comment, marked as answer]
solved with recreating container using compose

**Reply by niccokunzmann** [reply]
Recreating helped me, too!

I also mapped the dkim volumes:

```
      - ./data/mailserver/opendkim:/tmp/docker-mailserver/opendkim/
      - ./data/mailserver/opendkim:/etc/opendkim/
```

See: https://github.com/docker-mailserver/docker-mailserver/issues/4431#issuecomment-2755385843

**Reply by polarathene** [reply]
@niccokunzmann you do not need the extra volumes, that is explained: https://github.com/docker-mailserver/docker-mailserver/issues/4431#issuecomment-2762353370

The docs do inform you [that you need to restart the DMS container](https://docker-mailserver.github.io/docker-mailserver/v15.1/config/best-practices/dkim_dmarc_spf/#generating-keys) and [our troubleshooting docs](https://docker-mailserver.github.io/docker-mailserver/v15.1/config/debugging/) are very clear about how to restart DMS properly with `--force-recreate`.

This will automatically have the same benefit of copying over the config and ensuring permissions and ownership is correct:

https://github.com/docker-mailserver/docker-mailserver/blob/fe1280a7dd518f2a7ef5f0589f58b01df835053b/target/scripts/startup/setup.d/dmarc_dkim_spf.sh#L13-L14

https://github.com/docker-mailserver/docker-mailserver/blob/fe1280a7dd518f2a7ef5f0589f58b01df835053b/target/scripts/startup/setup.d/dmarc_dkim_spf.sh#L23-L32
